packages/network-core/src/QueryBuilder.ts

```diff
@@ -0,0 +1,63 @@
+import { PureCollection, PureModel } from '@datx/core';
+import { INetwork } from './interfaces/INetwork';
+import { IQueryConfig } from './interfaces/IQueryConfig';
+import { IRequestDetails } from './interfaces/IRequestDetails';
+import { ISubrequest } from './interfaces/ISubrequest';
+import { Request } from './Request';
+
+export class QueryBuilder<
```

Comment:
Could this be an abstract class? https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-classes-and-members